src/opengl.c

```diff
@@ -449,6 +687,116 @@ bool glx_load_prog_main(const char *vshader_str, const char *fshader_str,
 	return true;
 }
 
+/**
+ * @brief Release binding of a texture.
+ */
+void glx_release_texture(session_t *ps attr_unused, glx_texture_t **pptex) {
+	glx_texture_t *ptex = *pptex;
+	// Release binding
+	if (ptex->texture) {
+		// log_info("Deleting texture wh(%d %d)", ptex->width, ptex->height);
+		glBindTexture(ptex->target, 0);
+		glDeleteTextures(1, &ptex->texture);
+	}
+	free(ptex);
+	*pptex = NULL;
+
+	gl_check_err();
+}
+
+/**
+ * Bind a X pixmap to an OpenGL texture.
+ */
+bool glx_bind_texture(session_t *ps attr_unused, glx_texture_t **pptex, int x, int y,
+                      int width attr_unused, int height attr_unused, bool repeat attr_unused) {
```

Comment:
Probably due to the way I coded this, I first copied the entire gl_blur path as a skeleton, commented out all the internal of the functions and marked everything as “unused” to avoid warnings and slowly uncommented sections until I achieved the required result.